main/lsp/LSPIndexer.cc

```diff
@@ -231,7 +226,24 @@ LSPIndexer::getTypecheckingPath(const LSPFileUpdates &edit,
         prodCategoryCounterInc("lsp.slow_path_reason", "new_file");
         return TypecheckingPath::Slow;
     }
-    return getTypecheckingPathInternal(edit.updatedFiles, evictedFiles);
+
+    LSPFileUpdates::FastPathFilesToTypecheckResult result;
+    auto path = getTypecheckingPathInternal(result, edit.updatedFiles, evictedFiles);
+    if (path == TypecheckingPath::Fast) {
+        // If we're taking the fast path, that means that there's relevant information in `result` that we should stash
+        // away for the typechecker.
+        for (auto &[fref, idx] : result.changedFiles) {
+            edit.fastPathChangedFiles.emplace_back(fref.data(*initialGS).path(), idx);
+        }
+
+        edit.fastPathUseIncrementalNamer = !result.changedSymbolNameHashes.empty();
+
+        for (auto fref : result.extraFiles) {
+            edit.fastPathExtraFiles.emplace_back(fref.data(*initialGS).path());
+        }
```

Comment:
Note that this all goes away if you pass in the `LSPFileUpdates` to getTypecheckingPathInternal:

1.  `fastPathChangedFiles` is already redundant. We already have the `edit`, the `edit.updatedFiles` already stores `core::File` objects, and those already store their `path`, so that stanza is just deleted entirely
2.  We're already storing `fastPathUseIncrementalNamer` on the edit. May as well just let `getTypecheckingPathInternal` set it.
3.  `result.extraFiles` is a `vector<FileRef>` that we never use—we immediately turn around and convert it to a `vector<string>`. Can't we construct the `vector<string>` directly and skip the intermediate?

Reply:
I'd love to remove this, by threading the updates through, but in the meantime have simplified the `FastPathFilesToTypecheckResult` struct a bit so that we're passing less data out.